src/annotator/integrations/image-text-layer.js

```diff
@@ -1,29 +1,152 @@
 import debounce from 'lodash.debounce';
 
 import { ListenerCollection } from '../../shared/listener-collection';
+import {
+  rectCenter,
+  rectsOverlapHorizontally,
+  rectsOverlapVertically,
+  unionRects,
+} from '../util/geometry';
 
 /**
- * @typedef Rect
- * @prop {number} left
- * @prop {number} right
- * @prop {number} top
- * @prop {number} bottom
+ * @typedef WordBox
+ * @prop {DOMRect[]} glyphs
+ * @prop {string} text
+ * @prop {DOMRect} rect - Bounding rectangle of all glyphs in word
  */
 
 /**
- * @param {Rect|null} a
- * @param {Rect} b
+ * @typedef LineBox
+ * @prop {WordBox[]} words
+ * @prop {DOMRect} rect - Bounding rectangle of all words in line
  */
-function mergeBoxes(a, b) {
-  if (!a) {
-    return b;
+
+/**
+ * @typedef ColumnBox
+ * @prop {LineBox[]} lines
+ * @prop {DOMRect} rect - Bounding rectangle of all lines in column
```

Comment:
Lines and columns currently tightly wrap the words and lines they contain, respectively. It may be helpful in future to "inflate" the lines and columns so that they fill as much of the page as possible without intersecting other lines/columns. This would allow better behavior when a text selection starts outside the visual boundaries of a line or column, as the browser would be better able to map the cursor position to a DOM location in the text layer.